include/seahorn/Expr/TypeChecker.hh

```diff
@@ -1,17 +1,66 @@
 #pragma once
-
-#include "seahorn/Expr/ExprApi.hh"
+#include "seahorn/Expr/ExprCore.hh"
+#include "seahorn/Expr/ExprOpSort.hh"
 
 namespace expr {
-class TypeChecker {
+
+class TypeChecker;
+
+class TypeCheckerHelper {
```

Comment:
Lots of comments are missing. Especially in this file. What is a `TypeCheckerHelper`, why there is a need both for `TypeChecker` and helper. How to extend type checker when new operators are added. What is special about terminals, etc.